File: src/node/SupercellFlash/transformation/ColorTransform.ts
import * as util from "util";

import { NATIVE_COLOR_TRANSFORM } from "../../../native";

export class ColorTransform extends NATIVE_COLOR_TRANSFORM {
  [Symbol.toStringTag](): string {
    return "ColorTransform";
  }

  [util.inspect.custom](depth: number): string {
    if (depth === 1) {
      return `<${this[Symbol.toStringTag]()}>`;
    } else {
      return `<${this[Symbol.toStringTag]()} redAdd: ${
        this.redAdd
      }, greenAdd: ${this.greenAdd}, blueAdd: ${this.blueAdd}, alpha: ${
        this.alpha
      }, redMul: ${this.redMul}, greenMul: ${this.greenMul}, blueMul: ${
        this.blueMul
      } >`;
    }
  }

  toJSON(): object {
    return {
      alpha: this.alpha,
      redAdd: this.redAdd,
      greenAdd: this.greenAdd,
      blueAdd: this.blueAdd,
      redMul: this.redMul,
      greenMul: this.greenMul,
      blueMul: this.blueMul,
    };
  }
}

File: bindings/SupercellFlashNapi/transformation/ColorTransform.cpp
#include "ColorTransform.hpp"

namespace scNapi {
    PROPERTY(
        ColorTransform,
        alpha,
        parent->alpha = ToNativeValue<float>(value),
        return ToJSValue(info, parent->alpha)
    );

    PROPERTY(
        ColorTransform,
        redAdd,
        parent->redAdd = ToNativeValue<uint8_t>(value),
        return ToJSValue(info, parent->redAdd)
    );

    PROPERTY(
        ColorTransform,
        greenAdd,
        parent->greenAdd = ToNativeValue<uint8_t>(value),
        return ToJSValue(info, parent->greenAdd)
    );

    PROPERTY(
        ColorTransform,
        blueAdd,
        parent->blueAdd = ToNativeValue<uint8_t>(value),
        return ToJSValue(info, parent->blueAdd)
    );

    PROPERTY(
        ColorTransform,
        redMul,
        parent->redMul = ToNativeValue<float>(value),
        return ToJSValue(info, parent->redMul)
    );

    PROPERTY(
        ColorTransform,
        greenMul,
        parent->greenMul = ToNativeValue<float>(value),
        return ToJSValue(info, parent->greenMul)
    );

    PROPERTY(
        ColorTransform,
        blueMul,
        parent->blueMul = ToNativeValue<float>(value),
        return ToJSValue(info, parent->blueMul)
    );
}